From b33ef3e2cf928ef67ee77003861317dabd44df19 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Tue, 4 Dec 2012 15:01:51 -0800
Subject: [PATCH 164/795] MR1: CDH-9273. Fix the bug caused by CDH-8955 (ContextFactory doesn't reflect other changes)

Reason: Bug fix (unit tests fix)
Ref: CDH-9273
Author: Karthik Kambatla
(cherry picked from commit 1a75260397da059848589e7125e976b452c64287)
(cherry picked from commit c0c0840fd0bca02ff8653ec1c7213cc05f4b2607)
---
 .../apache/hadoop/mapreduce/ContextFactory.java    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapreduce/ContextFactory.java b/src/mapred/org/apache/hadoop/mapreduce/ContextFactory.java
index fd86426..51adf75 100644
--- a/src/mapred/org/apache/hadoop/mapreduce/ContextFactory.java
+++ b/src/mapred/org/apache/hadoop/mapreduce/ContextFactory.java
@@ -123,7 +123,7 @@
         WRAPPED_CONTEXT_FIELD = null;
       }
       MAP_CONTEXT_CONSTRUCTOR.setAccessible(true);
-      REPORTER_FIELD = taskIOContextCls.getDeclaredField("reporter");
+      REPORTER_FIELD = taskContextCls.getDeclaredField("reporter");
       REPORTER_FIELD.setAccessible(true);
       READER_FIELD = mapContextCls.getDeclaredField("reader");
       READER_FIELD.setAccessible(true);
-- 
1.7.0.4

